src/App.tsx

```diff
@@ -1,39 +1,60 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
+
+interface State {
+  clockName: string,
+  hasClock: boolean,
+}
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('click', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: true });
+    });
+  }
```

Comment:
create functions for this code, and use it in both and and remove listeners
```suggestion
    document.addEventListener('click', this.handleClick)
  }
```

Reply:
you don't need to prevent default in 'click' event